tomcat/src/main/java/nextstep/jwp/ResourceReader.java

```diff
@@ -0,0 +1,33 @@
+package nextstep.jwp;
+
+import javassist.NotFoundException;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.File;
+import java.io.IOException;
+import java.net.URISyntaxException;
+import java.net.URL;
+import java.nio.file.Files;
+
+public class ResourceReader {
+
+    private final Logger log = LoggerFactory.getLogger(ResourceReader.class);
+
+    public String read(final String target) throws NotFoundException, URISyntaxException {
+        final URL resource = getClass().getClassLoader().getResource("static" + target);
+        if (resource == null) {
+            throw new NotFoundException("자원을 찾지 못했음 : " + target);
+        }
+        return readFile(new File(resource.toURI()));
+    }
+
+    private String readFile(final File file) {
+        try {
+            return new String(Files.readAllBytes(file.toPath()));
+        } catch (IOException e) {
+            log.error(e.getMessage());
+        }
+        return null;
+    }
+}
```

Comment:
null을 반환하고 있네요💀. null을 반환하면 어떤 문제가 발생할까요? [모던 자바 11장 - null 대신 Optional 클래스](https://iskull-dev.tistory.com/275?category=996541)를 참고해 보면 좋을거 같아요😃.
```suggestion
private Optional<String> readFile(final File file) {
        try {
            return new String(Files.readAllBytes(file.toPath()));
        } catch (IOException e) {
            log.error(e.getMessage());
        }
        return Optional.empty();
    }
```

Reply:
좋은 지적입니다!! 다만 현재 코드에서는 Optional보다 Exception을 반환하는 것이 예외 처리에 용이하여 예외를 반환하는 형태로 변경하였습니다!